lib/govuk_design_system_formbuilder/elements/datetime.rb

```diff
@@ -0,0 +1,348 @@
+# frozen_string_literal: true
+
+require "govuk_design_system_formbuilder"
+
+module GOVUKDesignSystemFormBuilder
+  module Elements
+    class DateTime < Base
+      using PrefixableArray
+
+      include Traits::Error
+      include Traits::Hint
+      include Traits::Supplemental
+      include Traits::HTMLClasses
+
+      SEGMENTS           = { day: "3i", month: "2i", year: "1i", hour: "4i", minute: "5i", offset: "8i" }.freeze
+      MULTIPARAMETER_KEY = { day: 3, month: 2, year: 1, hour: 4, minute: 5, offset: 8 }.freeze
+
+      def initialize(builder, object_name, attribute_name, legend:, caption:, hint:, omit_day:, maxlength_enabled:,
+                     form_group:, date_of_birth: false, **kwargs, &block)
+        super(builder, object_name, attribute_name, &block)
+
+        @legend            = legend
+        @caption           = caption
+        @hint              = hint
+        @date_of_birth     = date_of_birth
+        @omit_day          = omit_day
+        @maxlength_enabled = maxlength_enabled
+        @form_group        = form_group
+        @html_attributes   = kwargs
+      end
+
+      def html
+        Containers::FormGroup.new(*bound, **@form_group, **@html_attributes).html do
+          Containers::Fieldset.new(*bound, **fieldset_options).html do
+            safe_join([supplemental_content, hint_element, error_element, date])
+          end
+        end
+      end
+
+      private
+
+      def fieldset_options
+        { legend: @legend, caption: @caption, described_by: [error_id, hint_id, supplemental_id],
+          data:   { controller: "datetimes" } }
+      end
+
+      def date
+        tag.div(class: %(#{brand}-date-input)) do
+          safe_join([day, month, year, hour, minute, meridian, offset])
+        end
+      end
+
+      def omit_day?
+        @omit_day
+      end
```

Comment:
Not sure we'd want to support this - feels like date-time should always have a date.